File: sdks/go/anthropic.go
package amlexia

type AnthropicUsage struct {
	InputTokens  int
	OutputTokens int
}

func TrackAnthropicMessage(c *Client, model string, statusCode, latencyMs int, usage AnthropicUsage, costUSD *float64) error {
	est := EstimateEventCostUSD(costUSD, model, "anthropic", usage.InputTokens, usage.OutputTokens, 0)
	prov := "anthropic"
	return c.Track(Event{
		Endpoint:   "/v1/messages",
		Method:     "POST",
		StatusCode: statusCode,
		LatencyMs:  latencyMs,
		CostUSD:    est,
		Provider:   &prov,
		ModelName:  &model,
		TokensIn:   &usage.InputTokens,
		TokensOut:  &usage.OutputTokens,
	})
}

File: sdks/ruby/lib/amlexia/anthropic.rb
# frozen_string_literal: true

module Amlexia
  module Anthropic
    module_function

    def track_anthropic_message(client, model:, status_code:, latency_ms:, usage: {}, cost_usd: nil, **kwargs)
      event = Cost.enrich_event(
        cost_usd: cost_usd,
        model_name: model,
        provider: "anthropic",
        tokens_input: usage[:input_tokens],
        tokens_output: usage[:output_tokens]
      )
      client.track(
        endpoint: "/v1/messages",
        method: "POST",
        status_code: status_code,
        latency_ms: latency_ms,
        provider: "anthropic",
        model_name: model,
        tokens_input: usage[:input_tokens],
        tokens_output: usage[:output_tokens],
        cost_usd: event[:cost_usd],
        **kwargs
      )
    end
  end
end
